modules/ding_app_content_rss/views/ding_app_content_rss_handler_sales_status.inc

```diff
@@ -0,0 +1,124 @@
+    elseif (!$event->active) {
+      $ticket_link_status = 'closed';
+    }
+    elseif ($state === 'published') {
+      // If event is published we first check to see if there's any available
+      // tickets at all. If not the event is sold out and we can avoid fetching
+      // ticket types and sale periods. There might be a waiting-list so we
+      // check for that first before setting sold out.
+      if (empty($available_tickets)) {
+        if (!empty($event->waiting_list)) {
+          $ticket_link_status = 'waiting-list';
+        }
+        else {
+          $ticket_link_status = 'sold-out';
+        }
+      }
+      else {
+        // Else we have no other options than to fetch the tickets and look
+        // for open or upcoming sale periods.
+        $sale_periods = [];
+        foreach ($p2b->getPrices($event_maker_id, $event_id) as $price) {
+          $sale_periods[] = [
+            'begin' => strtotime($price->sale_begin_at),
+            'end' => strtotime($price->sale_end_at),
+          ];
+        }
+
+        // Some events, for example those migrated from the old API, will have
+        // sales period on the event and not on the individual tickets.
+        if (isset($event->sale_open_at) && isset($event->sale_close_at)) {
+          $sale_periods[] = [
+            'begin' => strtotime($event->sale_open_at),
+            'end' => strtotime($event->sale_close_at),
+          ];
+        }
+
+        $now = time();
+
+        // Filter out periods which are already over.
+        $sale_periods = array_filter($sale_periods, function ($p) use ($now) {
+          return $p['end'] >= $now;
+        });
+
+        // Check if we still have any active sale periods. If such are - go on!
+        if (!empty($sale_periods)) {
+          // Find the closest sale period.
+          $begin_values = array_column($sale_periods, 'begin');
+          // Get the closest sale period. In case when event has upcoming sales,
+          // its start value will be used in button label.
+          $min_period_begin = min($begin_values);
+          $min_period = array_filter($sale_periods, function ($sale_period) use ($min_period_begin) {
+            return ($sale_period['begin']) == $min_period_begin;
+          });
+          $min_period = array_shift($min_period);
+
+          // We have found an open sales period.
+          $ticket_link_status = 'open';
+
+          // Check for 'upcoming' sale.
+          if ($now < $min_period['begin']) {
+            // We haven't found any open sales period yet, but at least there's
+            // an upcoming, so set that status for now.
+            $ticket_link_status = 'upcoming';
+          }
+        }
+
+        // If no upcoming or open sales period was found for the published event
+        // we consider the sales status to be closed.
+        if (empty($ticket_link_status)) {
+          $ticket_link_status = 'closed';
+        }
+
+      }
+    }
+
+    return $ticket_link_status;
```

Comment:
Is this code copied from the .module file? Maybe we should move the status checking code into its own function?

Also I think a `module_exists('ding_place2book')` check would be good to have and return early if it's not installed.